File: data/src/main/java/com/everis/data/repository/ContentProviderDataRepository.java
package com.everis.data.repository;

import android.content.ContentResolver;
import android.database.Cursor;
import android.provider.ContactsContract;

import com.everis.data.model.LocalContactEntity;
import com.everis.data.model.LocalContactEntityDataMapper;
import com.everis.data.network.local.SpikeContentProvider;
import com.everis.domain.model.LocalContact;
import com.everis.domain.repository.ContentProviderRepository;

import java.util.ArrayList;
import java.util.List;

import rx.Observable;
import rx.Subscriber;
import rx.functions.Func1;

/**
 * Created by everis on 3/05/18.
 */

import android.content.Context;

public class ContentProviderDataRepository implements ContentProviderRepository {

    private SpikeContentProvider database;

    public ContentProviderDataRepository(SpikeContentProvider database) {
        this.database = database;
    }

    @Override
    public Observable<List<LocalContact>> getLocalContactsBatch() {
        return database.getLocalContatsBatch().map(new Func1<List<LocalContactEntity>, List<LocalContact>>() {
            @Override
            public List<LocalContact> call(List<LocalContactEntity> localContactEntities) {

                List<LocalContact> localContacts = new ArrayList<>();

                for(LocalContactEntity contactEntity : localContactEntities){
                    LocalContact localContact = LocalContactEntityDataMapper.transform(contactEntity);
                    localContacts.add(localContact);
                }

                return localContacts;
            }
        });
    }
}

File: data/src/main/java/com/everis/data/repository/ContentProviderDataRepository.kt
package com.everis.data.repository
import com.everis.data.model.LocalContactEntityDataMapper
import com.everis.data.network.local.SpikeContentProvider
import com.everis.domain.model.LocalContact
import com.everis.domain.repository.ContentProviderRepository

import java.util.ArrayList

import rx.Observable

class ContentProviderDataRepository(private val database: SpikeContentProvider) : ContentProviderRepository {

    override fun getLocalContactsBatch(): Observable<List<LocalContact>> {
        return database.getLocalContactsBatch().map { localContactEntities ->
            val localContacts = ArrayList<LocalContact>()

            for (contactEntity in localContactEntities) {
                val localContact = LocalContactEntityDataMapper.transform(contactEntity)
                localContacts.add(localContact)
            }

            localContacts
        }
    }
}
